### DISCUSSION TITLE
Privacy toggles?

### DISCUSSION THREAD

**Discussion by foragerr** [Arize-ai/phoenix#2609] [Ideas]
I would like to offer my langchain-chatbot users strong privacy guarantees. When a conversation is marked private, I want to not trace that interaction at alldo not want to trace that interaction at all. 

When using auto-instrument, It would be nice to be able to set a flag in code to omit an interaction from tracing.

I see some discussion about filtering at the UI level after all traces have been logged, I haven't yet seen come across filtering at the trace collection level.

**Comment by mikeldking** [comment]
@foragerr I think this is currently possible, I believe all our instrumentation respects this suppression strategy https://github.com/Arize-ai/phoenix/blob/main/src/phoenix/trace/__init__.py

See suppression code. 

Basically we respect suppression via a flag. Would this work for you?

**Reply by foragerr** [reply]
Sounds exactly like what I'm looking for. I will give it a shot, thank you.

**Comment by mikeldking** [comment]
This is now also available via `openinfernce-instrumentation` package so there's no need to use phoenix to suppress any openinference instrumentation at runtime.
